### DISCUSSION TITLE
setMapStoreSuffix() not working

### DISCUSSION THREAD

**Discussion by syntheticminds** [vuejs/pinia#1223] [Help and Questions] [answered]
### Reproduction

https://github.com/syntheticminds/pinia-bug

### Steps to reproduce the bug

Open _index.html_

### Expected behavior

_Value: 123_ seen in browser. No errors in console.

### Actual behavior

Blank page. Error in console caused by `this.counter_store` being undefined.

### Additional information

I suspect this is to do with Laravel Mix (Webpack) messing with `setMapStoreSuffix()`'s scope.

**Comment by posva** [comment, marked as answer]
Maybe it is related to laravel-mix. Have you tried moving the `setStoreSuffix()` directly to your `main.js`? You have to make sure `setStoreSuffix()` is called **before any `useMyStore()`**. e.g. call it right next to `app.use(pinia)`

**Reply by syntheticminds** [reply]
No luck. JS isn't my first language so I'm afraid I can't give much insight into the cause.

Obviously I can work around this but I thought you might like to know the way settings are injected may be incompatible with some build tools.

**Reply by 3zzy** [reply]
> Maybe it is related to laravel-mix. Have you tried moving the `setMapStoreSuffix()` directly to your `main.js`?

Can confirm this works.

**Reply by syntheticminds** [reply]
> Can confirm this works.

Still no luck with my test code: https://github.com/syntheticminds/pinia-bug

**Reply by FVolral** [reply]
I had the same problem. Figured out that you must use `setMapStoreSuffix()` where you will use `mapStores`. No sure if this is expected. It would be great to be able to set that suffix in `main.js` only once.  Here is the repro on codeSandBox where `setMapStoreSuffix` is working : https://codesandbox.io/s/pinia-setmapstoresuffix-bug-repro-lbfzit?file=/src/App.vue

**Reply by RedShift1** [reply]
I found that instead of relying on `setMapStoreSuffix`, you can also do something like this in your Vue 2 component (using options API):

```javascript
const store = defineStore('$store2', { ... });

export default
{
    computed:
    {
        $store2() { return store(this.$pinia) }
    },
    ...
}
```
